messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/engine/EngineWebViewInterface.kt

```diff
@@ -15,13 +16,34 @@ internal data class EngineWebEvent(
     val parameters: Map<String, Any>? = null
 )
 
-class EngineWebViewInterface constructor(
-    listener: EngineWebViewListener
-) {
+class EngineWebViewInterface(listener: EngineWebViewListener) {
     private var listener: EngineWebViewListener = listener
 
+    // Indicates whether the interface is attached to a web view and should continue to process messages
+    private var isAttachedToWebView: Boolean = false
+
+    internal fun attach(webView: WebView) {
+        // Indicates that the interface is attached to WebView and should start processing messages
+        isAttachedToWebView = true
+        // Add JavaScript interface to WebView so that it can communicate with the interface
+        webView.addJavascriptInterface(this, JAVASCRIPT_INTERFACE_NAME)
+    }
+
+    internal fun detach(webView: WebView) {
+        // Indicates that the interface is no longer attached to WebView and should stop processing messages
+        isAttachedToWebView = false
+        // Remove JavaScript interface from WebView
+        webView.removeJavascriptInterface(JAVASCRIPT_INTERFACE_NAME)
+    }
+
     @JavascriptInterface
     fun postMessage(message: String) {
+        // Check if the interface is attached to WebView
+        // If not, do not process the message
+        if (!isAttachedToWebView) {
```

Comment:
`removeJavascriptInterface` method doesn't work as I expected and keeps sending messages here. So I added flag to prevent sending calls to native when we know that the message has been cancelled.